src/Data/Tree/AVL/Indexed/Relation/Unary/Any.agda

```diff
@@ -0,0 +1,118 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- AVL trees where at least one element satisfies a given property
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Relation.Binary using (StrictTotalOrder)
+
+module Data.Tree.AVL.Indexed.Relation.Unary.Any
+  {a ℓ₁ ℓ₂} (strictTotalOrder : StrictTotalOrder a ℓ₁ ℓ₂)
+  where
+
+open import Data.Nat.Base using (ℕ)
+open import Data.Product using (_,_; ∃; -,_; proj₁; proj₂)
+open import Data.Sum.Base using (_⊎_; inj₁; inj₂; [_,_]′)
+open import Function.Base
+open import Level using (Level; _⊔_)
+
+open import Relation.Nullary using (Dec; no)
+open import Relation.Nullary.Decidable using (map′)
+open import Relation.Nullary.Sum using (_⊎-dec_)
+open import Relation.Unary
+
+open StrictTotalOrder strictTotalOrder renaming (Carrier to Key)
+open import Data.Tree.AVL.Indexed strictTotalOrder as AVL
+  using (Tree; Value; Key⁺; [_]; _<⁺_; K&_; _,_; key; _∼_⊔_; ∼0; leaf; node)
+
+
+private
+  variable
+    v p q : Level
+    V : Value v
+    P : Pred (K& V) p
+    Q : Pred (K& V) q
+    l u : Key⁺
+    n : ℕ
+    t : Tree V l u n
+
+------------------------------------------------------------------------
+-- Definition
+
+-- Given a predicate P, Any P t is a path to one element in t that satisfies P.
+-- There may be others.
+-- See `Relation.Unary` for an explanation of predicates.
+
+data Any {V : Value v} (P : K& V → Set p) {l u}
+     : ∀ {n} → Tree V l u n → Set (p ⊔ a ⊔ v ⊔ ℓ₂) where
+  here  : ∀ {hˡ hʳ h} {kv : K& V} → P kv →
+          {lk : Tree V l [ kv .key ] hˡ}
+          {ku : Tree V [ kv .key ] u hʳ}
+          {bal : hˡ ∼ hʳ ⊔ h} →
+          Any P (node kv lk ku bal)
+  left  : ∀ {hˡ hʳ h} {kv : K& V}
+          {lk : Tree V l [ kv .key ] hˡ} →
+          Any P lk →
+          {ku : Tree V [ kv .key ] u hʳ}
+          {bal : hˡ ∼ hʳ ⊔ h} →
+          Any P (node kv lk ku bal)
+  right : ∀ {hˡ hʳ h} {kv : K& V}
+          {lk : Tree V l [ kv .key ] hˡ}
+          {ku : Tree V [ kv .key ] u hʳ} →
+          Any P ku →
+          {bal : hˡ ∼ hʳ ⊔ h} →
+          Any P (node kv lk ku bal)
+
+------------------------------------------------------------------------
+-- Operations on Any
+
+map : P ⊆ Q → Any P t → Any Q t
+map f (here  p) = here (f p)
+map f (left  p) = left (map f p)
+map f (right p) = right (map f p)
+
+lookup : {t : Tree V l u n} → Any P t → Key
```

Comment:
Why not define a `lookup` function returning a key and value here? This function could then be defined in terms of it and called `lookupKey`...